user/pages/03.commerce2/02.developer-guide/09.promotions/02.create-a-condition/docs.md

```diff
@@ -4,4 +4,222 @@ taxonomy:
+
+  /** * Gets the condition label. */ 
+  public function getLabel();
+
+  /** * Shown in the condition UI when enabling/disabling a condition. */
+  public function getDisplayLabel();
+
+  /** * This is the entity type ID of the entity passed to evaluate(). */
+  public function getEntityTypeId();
+
+  /**
+   * Evaluates the condition.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity.
+   *
+   * @return bool
+   *   TRUE if the condition has been met, FALSE otherwise.
+   */
+  public function evaluate(EntityInterface $entity);
+}
+
+```
+[ConditionInterface.php](https://cgit.drupalcode.org/commerce/tree/src/Plugin/Commerce/Condition/ConditionInterface.php)
+
+```php
+<?php
+
+/**
+ * Provides the base class for conditions.
+ */
+abstract class ConditionBase extends PluginBase implements ConditionInterface {
+
+  public function __construct(array $configuration, $plugin_id, $plugin_definition) { }
+
+  public function calculateDependencies() { ... }
+
+  public function defaultConfiguration() { ... }
+
+  public function getConfiguration() { ... }
+
+  public function setConfiguration(array $configuration) { ... }
+
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) { ... }
+
+  public function validateConfigurationForm(array &$form, FormStateInterface $form_state) {}
+
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) { ... }
+
+  // These are required by ConditionInterface.
+  public function getLabel() { ... }
+
+  public function getDisplayLabel() { ... }
+
+  public function getEntityTypeId() { ... }
+  
+  // TODO function evaluate is missing, write your own.
+}
+```
+[ConditionBase.php](https://cgit.drupalcode.org/commerce/tree/src/Plugin/Commerce/Condition/ConditionBase.php)
+
+
+Functionally, a condition would not be useful without some sort of user 
+configuration. defaultConfiguration, buildConfigurationForm, and 
+validateConfigurationForm go hand in hand with evaluate to create a basic 
+condition.
+
+Lets see how these functions are used in practice by analysing one of 
+the simplest conditions to ship with commerce: OrderItemQuantity.
+
+
+Example using class OrderItemQuantity:
+
+User story:
+  As a store manager I want to encourage bulk ordering by creating a promotion that applies to orders of at least 5 of the selected product. 
+  
+ ![Add Promotion UI](create-a-promotion.png)
+  
+
+```php
+<?php // namespace & use
+/**
+ * Provides the total discounted product quantity condition.
+ *
+ * Implemented as an order condition to be able to count products across
+ * non-combined order items.
+ *
+ * @CommerceCondition(
+ *   id = "order_item_quantity",
+ *   label = @Translation("Total discounted product quantity"),
+ *   category = @Translation("Products"),
+ *   entity_type = "commerce_order",
+ *   parent_entity_type = "commerce_promotion",
+ *   weight = 10,
+ * )
+ */
```

Comment:
Let's break this up and talk about the parts of it. Maybe separate it into a section on the annotations, the configuration form, and then the evaluate method. That would let readers digest it in smaller pieces.

Reply:
I suspect talking a lot more about the parts of a condition using this example would allow you to drop most or all of the ConditionBase and ConditionInterface code from the docs.